packages/dds/tree/src/test/simple-tree/simpleTree.bench.ts

```diff
@@ -80,6 +81,322 @@ describe("SimpleTree benchmarks", () => {
 				},
 			});
 		}
+
+		describe("Access to leaves", () => {
+			describe("Optional object property", () => {
+				const factory = new SchemaFactory("test");
+				class MyInnerSchema extends factory.object("inner", {
+					value: factory.optional(factory.number),
+				}) {}
+				class MySchema extends factory.object("root", {
+					value: factory.optional(factory.number),
+					leafUnion: factory.optional([factory.number, factory.string]),
+					complexUnion: factory.optional([factory.number, MyInnerSchema]),
+				}) {}
+				let tree: MySchema;
+				let readNumber: number | undefined;
+
+				benchmark({
+					type: BenchmarkType.Measurement,
+					title: `Read value from leaf`,
+					before: () => {
+						tree = new MySchema({ value: 1 });
```

Comment:
These tests (which are great btw) seems to all use unhydrated nodes meaning they don't use the actual flex-tree. Consider constructing the nodes with out "hydrate" test utility/function to test with real flex-tree nodes. I suspect the performance might be significantly different.

Might actually be worth testing both cases if you can do it cleanly (make loop over the tests twice with different factory functions?)